src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/QuantitySearchValue.cs

```diff
@@ -63,14 +64,21 @@ public static QuantitySearchValue Parse(string s)
                 throw new FormatException(Core.Resources.MoreThanTwoTokenSeparatorSpecified);
             }
 
-            decimal quantity = decimal.Parse(parts[0], NumberStyles.Number, CultureInfo.InvariantCulture);
-            string system = parts.Count > 1 ? parts[1] : string.Empty;
-            string code = parts.Count > 2 ? parts[2] : string.Empty;
-
-            return new QuantitySearchValue(
-                system.UnescapeSearchParameterValue(),
-                code.UnescapeSearchParameterValue(),
-                quantity);
+            try
+            {
+                decimal quantity = decimal.Parse(parts[0], NumberStyles.Number, CultureInfo.InvariantCulture);
+                string system = parts.Count > 1 ? parts[1] : string.Empty;
+                string code = parts.Count > 2 ? parts[2] : string.Empty;
+
+                return new QuantitySearchValue(
+                    system.UnescapeSearchParameterValue(),
+                    code.UnescapeSearchParameterValue(),
+                    quantity);
+            }
+            catch
+            {
+                throw new BadRequestException($"Malformed search value \"{parts[0]}\"");
```

Comment:
This text should be in the corresponding resource file.

Reply:
Oh - I missed that. This has been fixed. Now using `Core.Resources.MalformedSearchValue`